lib/authorizers/websocket-authorizer.js

```diff
@@ -0,0 +1,50 @@
+var util = require('util'),
+    BaseAuthorizer = require('./base-authorizer').BaseAuthorizer,
+    AuthorizerFailedError = require('./base-authorizer').AuthorizerFailedError;
+
+// Given a websocket, authenticate it using our protocol
+// when authorize is called
+function WebsocketAuthorizer(ws) {
+  this._ws = ws;
+  this._authorized = false;
+};
+
+util.inherits(WebsocketAuthorizer, BaseAuthorizer);
+
+// Should be overriden in Authorizers
+// 
+// expects 
+// apiKey:string - the apiKey to authorize with
+// cb:function - a function that takes (err, token) after authorization occurs
+WebsocketAuthorizer.prototype.authorize = function(apiKey, cb) {
+  if (!this._ws) return cb(new AuthorizerFailedError("websocket not valid"));
+  if (this._ws.readyState != 1) return cb(new AuthorizerFailedError("websocket not open. state: "+this._ws.readyState));
+  
+  var self = this;

```

Comment:
interesting; i can do that, but it's a different pattern then the rest of the code (the self pattern is "needed" elsewhere, so i wanted to use the same pattern). let me know your thoughts on this